imx.mk

```diff
@@ -0,0 +1,198 @@
+	$(TF_A_EXPORTS) $(MAKE) -C $(TF_A_PATH) $(TF_A_FLAGS) all fip
+
+arm-tf-clean:
+	$(TF_A_EXPORTS) $(MAKE) -C $(TF_A_PATH) $(TF_A_FLAGS) clean
+
+################################################################################
+# U-Boot
+################################################################################
+
+U-BOOT_EXPORTS = CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+
+U-BOOT_DEFCONFIG_FILES := $(U-BOOT_PATH)/configs/imx8mq_evk_defconfig \
+			  $(BUILD_PATH)/kconfigs/uboot_imx8.conf
+
+$(U-BOOT_PATH)/.config: $(U-BOOT_DEFCONFIG_FILES)
+	$(U-BOOT_EXPORTS) $(MAKE) -C $(U-BOOT_PATH) imx8mq_evk_defconfig
+	(cd $(U-BOOT_PATH) && \
+                ARCH=arm64 \
+                scripts/kconfig/merge_config.sh $(U-BOOT_DEFCONFIG_FILES))
+
+.PHONY: u-boot-defconfig
+u-boot-defconfig: $(U-BOOT_PATH)/.config
+
+# The imx8mq-evk.dtb in the symlink below needs to be copied with prefix
+# "fsl-", since the imx-mkimage later on refers to the dtb with that prefix.
+.PHONY: u-boot
+u-boot: u-boot-defconfig
+	$(U-BOOT_EXPORTS) $(MAKE) -C $(U-BOOT_PATH)
+
+.PHONY: u-boot-clean
+u-boot-clean:
+	cd $(U-BOOT_PATH) && git clean -xdf
+
+.PHONY: u-boot-cscope
+u-boot-cscope:
+	$(U-BOOT_EXPORTS) $(MAKE) -C $(U-BOOT_PATH) cscope
+
+
+################################################################################
+# Linux kernel
+################################################################################
+LINUX_DEFCONFIG_COMMON_ARCH := arm64
+LINUX_DEFCONFIG_COMMON_FILES := \
+		$(LINUX_PATH)/arch/arm64/configs/defconfig \
+		$(CURDIR)/kconfigs/imx.conf
+
+linux-defconfig: $(LINUX_PATH)/.config
+
+LINUX_COMMON_FLAGS += ARCH=arm64
+
+linux: linux-common
+
+linux-defconfig-clean: linux-defconfig-clean-common
+
+LINUX_CLEAN_COMMON_FLAGS += ARCH=arm64
+
+linux-clean: linux-clean-common
+
+LINUX_CLEANER_COMMON_FLAGS += ARCH=arm64
+
+linux-cleaner: linux-cleaner-common
+
+################################################################################
+# OP-TEE
+################################################################################
+optee-os: optee-os-common
+optee-os-clean: optee-os-clean-common
+
+################################################################################
+# DDR firmware
+################################################################################
+# This is prebuilt binaries by NXP, download them and use them. Update path if
+# it changes in the future.
+
+$(ROOT)/out-firmware/$(FIRMWARE_BIN):
+	mkdir -p $(ROOT)/out-firmware
+	(cd $(ROOT)/out-firmware && wget $(FIRMWARE_BIN_URL))
+
+$(FIRMWARE_PATH)/.unpacked: $(ROOT)/out-firmware/$(FIRMWARE_BIN)
+	(cd $(ROOT)/out-firmware && \
+	 echo $(FIRMWARE_BIN_SHA256_SUM) $(FIRMWARE_BIN) | sha256sum -c)
+	(cd $(ROOT)/out-firmware && \
+	 chmod 711 $(FIRMWARE_BIN) && ./$(FIRMWARE_BIN) --auto-accept)
+	touch $(FIRMWARE_PATH)/.unpacked
+	
+.PHONY: ddr-firmware
+ddr-firmware: $(FIRMWARE_PATH)/.unpacked
+
+ddr-firmware-clean:
+	rm -rf $(ROOT)/out-firmware
```

Comment:
First time I'm seeing this in a Makefile / script from you :) ... I very much OK with it, myself! :)

Reply:
Yeah, it's not good but there wasn't that many options here.